src/main/java/com/microsoft/sqlserver/jdbc/PerformanceLog.java

```diff
@@ -0,0 +1,45 @@
+package com.microsoft.sqlserver.jdbc;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class PerformanceLog {
+
+    static final java.util.logging.Logger perfLoggerConnection = java.util.logging.Logger
+            .getLogger("com.microsoft.sqlserver.jdbc.PerformanceMetrics.Connection");
+
+    //TODO
+    //More loggers to be added here e.g. com.microsoft.sqlserver.jdbc.PerformanceMetrics.Statement
+    
+    public static class Scope implements AutoCloseable {
+        private Logger logger;
+        private String logPrefix;
+        private PerformanceActivity activity;
+        private long startTime;
+        private final boolean enabled;
+
+        public Scope(Logger logger, String logPrefix, PerformanceActivity activity) {
+            this.enabled = logger.isLoggable(Level.INFO);
+            if (enabled) {
+                this.logger = logger;
+                this.logPrefix = logPrefix;
+                this.activity = activity;
+                this.startTime = enabled ? System.currentTimeMillis() : 0;

```

Comment:
This can be simplified since it's now with the if block:
```suggestion
                this.startTime = System.currentTimeMillis();
```